### PR TITLE
[fix](move-memtable) multi replica tables should tolerate minority failures

### PR DESCRIPTION
**Pull request by kaijchen**
## Proposed changes

Load job for multi replica tables shouldn't fail immediately on any single replica errors.
Errors should be recorded and reported for individual replica of tablets, and checked on commit info.

### CODE REVIEW

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

File: be/src/vec/sink/writer/vtablet_writer_v2.cpp [hunk 1]
```diff
@@ -568,23 +594,39 @@ Status VTabletWriterV2::close(Status exec_status) {
 
         // send CLOSE_LOAD on all non-incremental streams if this is the last sink
         if (is_last_sink) {
-            RETURN_IF_ERROR(_load_stream_map->close_load(false));
+            auto st = _load_stream_map->close_load(false);
+            if (!st.ok()) {
```
Comment (liaoxin01) [inline be/src/vec/sink/writer/vtablet_writer_v2.cpp:598]: We should handle the error here. The failure of downstream nodes should not be visible to VtabletWriter.

File: be/src/vec/sink/writer/vtablet_writer_v2.cpp [hunk 2]
```diff
@@ -568,23 +594,39 @@ Status VTabletWriterV2::close(Status exec_status) {
...
+            }
         }
 
         // close_wait on all non-incremental streams, even if this is not the last sink.
         // because some per-instance data structures are now shared among all sinks
         // due to sharing delta writers and load stream stubs.
-        RETURN_IF_ERROR(_close_wait(false));
+        {
+            auto st = _close_wait(false);
+            if (!st.ok()) {
```
Comment (liaoxin01) [inline be/src/vec/sink/writer/vtablet_writer_v2.cpp:608]: ditto

File: be/src/vec/sink/writer/vtablet_writer_v2.cpp [hunk 3]
```diff
@@ -651,9 +694,12 @@ Status VTabletWriterV2::_close_wait(bool incremental) {
...
                         return Status::TimedOut("load timed out before close waiting");
                     }
-                    RETURN_IF_ERROR(stream->close_wait(_state, remain_ms));
+                    auto st = stream->close_wait(_state, remain_ms);
+                    if (!st.ok() && status.ok()) {
+                        status = st;
+                    }
                 }
-                return Status::OK();
+                return status;
```
Comment (liaoxin01) [inline be/src/vec/sink/writer/vtablet_writer_v2.cpp:702]: Will there be a load_stram leak here? If there is an error, it will be returned.

Comment (kaijchen) [reply be/src/vec/sink/writer/vtablet_writer_v2.cpp:686]: This is inside lambda, see `LoadStreamMap::for_each_st()`.

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Review by sollhui — APPROVED** [review]
LGTM

**Pull request merged**